src/EventListener/StoreFbcSubscriber.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusFacebookPlugin\EventListener;
+
+use Symfony\Component\HttpKernel\KernelEvents;
+
+final class StoreFbcSubscriber extends AbstractSubscriber
+{
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            KernelEvents::REQUEST => 'storeFbc',
+        ];
+    }
+
+    public function storeFbc(): void
+    {
+        if (!$this->isRequestEligible()) {
+            return;
+        }
+
+        $request = $this->requestStack->getCurrentRequest();
+        if (null === $request) {
+            return;
+        }
+
+        $fbclid = $request->query->get('fbclid');
+        if (!is_string($fbclid)) {
+            // No sense to proceed, we can generate fbc from fbclid only
+            return;
+        }
+
+        $session = $request->getSession();
+        /** @var string|mixed $fbc */
+        $fbc = $session->get('ssfp_fbc');
+        if (is_string($fbc)) {
+            // We already have fbc at session
+            // Extract fbclid from fbc to compare
+            $existingFbclid = $this->extractFbclid($fbc);
+            if ($existingFbclid === $fbclid) {
+                // If fbclid is the same - we shouldn't regenerate fbc
+                // and use old one with old timestamp
+                return;
+            }
+        }
+
+        $creationTime = time();
```

Comment:
It says in the docs it should be milliseconds: https://developers.facebook.com/docs/marketing-api/conversions-api/parameters/fbp-and-fbc.
> creationTime is the UNIX time since epoch in milliseconds when the _fbc cookie was saved. If you don't save the _fbc cookie, use the timestamp when you first observed or received this fbclid value.